### ISSUE TITLE
Should only try update when device is online

### ISSUE THREAD

**Issue by Discostu36** [Tobi823/ffupdater#105]
Ffupdater gives me an error every night that updates failed because of connectivity issues. My phone is in airplane mode at night. Ffupdater should check online status of the device before looking for updates. Only when updates fail despite the device being online there should be an error.

**Comment by Tobi823** [comment]
> Ffupdater should check online status of the device before looking for updates.

FFUpdater already does this. But doing it reliably for all smartphones from all manufacturers is very difficult. Some devices simply do not behave as specified in the Android documentation. (See https://gist.github.com/Farbklex/f84029889444ee9c52a331a7e2bd10d2)

Apparently my logic does not work for your smartphone. Can you send me the exact error message (stacktrace) to understand the difference?

**Comment by Discostu36** [comment]
```
Stacktrace:
```
de.marmaro.krt.ffupdater.app.impl.exceptions.NetworkException: Fail to consume kotlin.String from 'https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.release.latest.arm64-v8a/artifacts/public/logs/chain_of_trust.log'.
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer.readNetworkResource(ApiConsumer.kt:61)
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer.access$readNetworkResource(ApiConsumer.kt:23)
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer$consumeNetworkResource$2.invokeSuspend(ApiConsumer.kt:38)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
Caused by: java.net.UnknownHostException: Unable to resolve host "firefox-ci-tc.services.mozilla.com": No address associated with hostname
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:156)
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:103)
	at java.net.InetAddress.getAllByName(InetAddress.java:1152)
	at com.android.okhttp.Dns$1.lookup(Dns.java:41)
	at com.android.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:178)
	at com.android.okhttp.internal.http.RouteSelector.nextProxy(RouteSelector.java:144)
	at com.android.okhttp.internal.http.RouteSelector.next(RouteSelector.java:86)
	at com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:176)
	at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:128)
	at com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:289)
	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:232)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:465)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getResponse(HttpURLConnectionImpl.java:411)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getHeaders(HttpURLConnectionImpl.java:167)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getHeaderField(HttpURLConnectionImpl.java:211)
	at java.net.URLConnection.getContentEncoding(URLConnection.java:551)
	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getContentEncoding(DelegatingHttpsURLConnection.java:143)
	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getContentEncoding(HttpsURLConnectionImpl.java:30)
	at de.marmaro.krt.ffupdater.app.impl.fetch.ApiConsumer.readNetworkResource(ApiConsumer.kt:49)
	... 8 more
Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
	at libcore.io.Linux.android_getaddrinfo(Native Method)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:73)
	at libcore.io.BlockGuardOs.android_getaddrinfo(BlockGuardOs.java:202)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:73)
	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:135)
	... 27 more
```
Device information:
| key | value |
| --- | --- |
| version name | 75.0.0 |
| version code | 91 |
| build type | release |
| SDK version | 30 |
| android release | 11 |
| board | FP4 |
| brand | Fairphone |
| device | FP4 |
| manufacturer | Fairphone |
| model | FP4 |
| product | FP4eea |
| SOC manufacturer | / |
| SOC model | / |
| supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| tags | release-keys |
| build time | 1639387934000 |
```

**Comment by Discostu36** [comment]
I am using NetGuard by the way, it was mentioned in your link as a possible cause of the problem.

**Comment by Tobi823** [comment]
Will be fixed in 75.2.0
